clean.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b         Delete bin directory"
+    echo "  -p         Delete packages directory"
+    echo "  -t         Delete Tools directory"
+    echo "  -w         Remove all untracked files in the working tree, including the bin, packages, and Tools directories"
+    echo "  -c         Delete the user-local NuGet package caches, located in ~/.local/share/dnu and ~/.local/share/NuGet"
+    echo "  -a, --all  Clean all of the above"
+    exit 1
+}
+
+invoke_buildproj()
+{
+    PackageRoot=$WorkingTreeRoot/packages
+    MSBuildPackageId="Microsoft.Build.Mono.Debug"
+    MSBuildPackageVersion="14.1.0.0-prerelease"
+    MSBuildPath=$PackageRoot/$MSBuildPackageId.$MSBuildPackageVersion/lib/MSBuild.exe
+
```

Comment:
Soon this will hopefully not depend on mono but instead msbuild on .NET Core. cc @joperezr